job-roles/principal-engineer.md

```diff
@@ -0,0 +1,41 @@
+# Principal Software Engineer
+
+{{#include intro.md}}
+
+## In the nutshell
+
+You will work under minimal direction to design and develop high performance storage algorithms for the next generation of NVMe over Fabrics storage solutions in containers for containers. This is an opportunity to own critical algorithms that must be invented to match the performance of next generation solid state storage technologies. The Principal Engineer must have the ability to discuss and analyze abstract algorithms with the team to evaluate and refine the design prior to implementation. Additionally, the Principal Engineer will be responsible for translating the design to an implementation in Rust/C/C++ with a clear understanding of practical system issues. The Principal Engineer will have the technical ability to write functional specs as well as responding to requirement documents and system level test plans, propose design changes and suggestions to team, collaborate with engineers and other product groups as needed cross functionally.
+
+We’re looking for a deeply skilled Principal Software Engineer to work closely with the product and the open source community to build the Mayadata vision. You'll join a highly collaborative team working along with talented engineers focusing to be the best in Container Attached Storage.
+
+## Role Description:
+
+- System engineering and implementation in Rust and C.
+- Working on performance issues using creative experiments and internally developed product features.
+- Research, propose, and integrate relevant open-source projects based on product objectives.
+- Write organized, efficient, and well documented Rust and C/C++ code as an example for junior engineers.
+- Participation in all levels of product definition, design, implementation, testing, and deployment.
+- Must include the ability to discuss abstract system architectures from ideas through implementation and creatively apply domain experience to solve technical challenges.
+- Mentoring software engineers, fostering an environment of trust and accountability.
+
+## Requirements:
+
+- Programming experience collaborating on software development Open Source Community.
+- Programming in C or C++ or Rust, or other systems programming languages.
+- Capable of meeting tight deadlines with excellent overall project and product development life cycle experience and team spirit.
+- Experience in agile and scrum methodologies.
+- Linux OS internals experience.
+- Excellent knowledge of multithreading, thread-pools with strong OOP skills.
```

Comment:
Not sure, why are we explicitly mentioning strong OOP skills. I would argue that functional programming is more valuable for system engineering than OOP.

Reply:
Thank you for the suggestion. I have added it